### ISSUE TITLE
Halfling's Lucky feature with advantage/disadvantage rerolls all 1's

### ISSUE THREAD

**Issue by bryanwh** [kakaroto/Beyond20#540]
With the newly added support for the halfling's Lucky feature, when you roll with advantage/disadvantage and get a 1 on both dice, it rerolls both dice rather than only 1 according to the rules on Advantage and Disadvantage quoted below.

> When you have advantage or disadvantage and something in the game, such as the halfling's Lucky trait, lets you reroll or replace the d20, you can reroll or replace only one of the dice. You choose which one. For example, if a halfling has advantage or disadvantage on an ability check and rolls a 1 and a 13, the halfling could use the Lucky trait to reroll the 1. ([source](https://www.dndbeyond.com/sources/basic-rules/using-ability-scores#AdvantageandDisadvantage))

Here's a screenshot of this happening (on Roll20, not sure about other VTTs):
![Screenshot from 2020-10-04 22-08-24](https://user-images.githubusercontent.com/46338456/95014052-04c4dd80-0690-11eb-95cf-65938afffe8f.png)
![Screenshot from 2020-10-04 22-08-29](https://user-images.githubusercontent.com/46338456/95014054-07bfce00-0690-11eb-9ed5-87a6e0e116ce.png)

I suspect it might be a technical limitation with Roll20 as described [here](https://app.roll20.net/forum/post/5462940/halfling-luck-bug-in-the-roll20-5th-edition-ogl-sheet), but I'm not sure.

**Comment by Aeristoka** [comment]
Interesting catch for sure.

**Comment by kakaroto** [comment]
😬 yeah, interesting indeed, not sure how I could get this fixed... I wouldn't be affected by roll20's technical limitation but it would be a beyond20 technical limitation at this point...
Also, congratulations on rolling a double natural one. That didn't take long 🤣

**Comment by bryanwh** [comment]
I actually spammed rolls specifically to test this. I did get a double natural one in-game though, right before the update.

**Comment by Aeristoka** [comment]
> I actually spammed rolls specifically to test this. I did get a double natural one in-game though, right before the update.

The pain... That is rough...

**Comment by bryanwh** [comment]
Yep, imagine getting a natural 1 with advantage. At least as a halfling I got to reroll. But in the same session, I also rerolled a 1 only to get another 1. This happened _twice_. So effectively 3 double nat ones in one session. I guess that means 3 double nat 20's next session?

**Comment by kakaroto** [comment]
Yeah, that's rough, I would suggest you put your PC in a box and buy a new one (like the 'dice prison' for unlucky dice).
